test/test_metapackage.py

```diff
@@ -76,6 +86,14 @@ def test_validate_metapackage(self):
                 else:
                     _validate_metapackage(path, package)
 
+    def test_collect_warnings(self):
+        """Tests warnings collection"""
+        warnings = []
+        pkgs_dict = find_packages(test_data_dir, warnings=warnings)
+
+        for w, we in zip(warnings, test_expected_warnings):
+            assert(w == we)
```

Comment:
You should try to give a second argument to `assert` when ever you use it, because this line is failing on the farm, but I don't know why exactly.

You can also just do `self.assertEqual(sorted(w), sorted(we))`, or if you are certain what the order should be `self.assertEqual(w, we)`. The `self.assertEqual` will automatically give you a nice comparison out if it fails.